### DISCUSSION TITLE
Caller information should also contain module information

### DISCUSSION THREAD

**Discussion by flenter** [autometrics-dev/autometrics-shared#42] [Ideas]
Right now the caller function can be stored under `caller` label.  The value of this label is set to the function as it appears in the code, however this doesn't make it unique (given that there's no module name included). 

We probably want to change this so we can ensure that when looking for functions that we're called by. another function actually match exactly (i.e. also matches on the module). In order to achieve this we could either.

* update the caller label to include function & module name in a single string
* add an additional field with something like `caller_module`
* rename the caller field and add a `caller_module` name

If people have better ideas, preferences, please let your ideas, suggestions & preferences be heard!

**Comment by emschwartz** [comment]
I think it probably makes sense to have `caller_function` and `caller_module` so that they would exactly match the values for `function` and `module`. If we add a [`class` label](https://github.com/orgs/autometrics-dev/discussions/28), we'd also want a `caller_class`.

**Comment by hatchan** [comment]
As I see it, adding these labels will not add to any cardinality, but will make sure that we accurately know which fn is calling/being called (in case of function name collisions). So think we should proceed with these changes, so we need to define it in the spec and implement it in the libraries.

I think that we can include these changes with the changes that we're already doing in https://github.com/orgs/autometrics-dev/discussions/33

**Reply by emschwartz** [reply]
Agreed. I'll write up the spec change without the `class` label for now and if we decide to add that, we can also make that change include having a `caller_class` label too.

**Comment by emschwartz** [comment]
The implementation of this is being tracked in https://github.com/autometrics-dev/autometrics-shared/issues/75
